extensions/smallrye-jwt/deployment/src/main/java/io/quarkus/smallrye/jwt/deployment/SmallryeJwtDevModeProcessor.java

```diff
@@ -0,0 +1,146 @@
+            }
+
+            devServices.produce(smallryeJwtDevServiceWith(devServiceProps));
+            return;
+        }
+
+        KeyPairContext ctx = liveReloadBuildItem.getContextObject(KeyPairContext.class);
+
+        LOGGER.info("The smallrye-jwt extension has configured an in-memory key pair, which is not enabled in production. " +
+                "Please ensure the correct keys/locations are set in production to avoid potential issues.");
+        if (ctx == null && !liveReloadBuildItem.isLiveReload()) {
+            // first execution
+            KeyPair keyPair = KeyUtils.generateKeyPair(KEY_SIZE);
+            String publicKey = getStringKey(keyPair.getPublic());
+            String privateKey = getStringKey(keyPair.getPrivate());
+
+            Map<String, String> devServiceProps = generateDevServiceProperties(publicKey, privateKey);
+
+            if (!isConfigPresent(MP_JWT_VERIFY_ISSUER) && !isConfigPresent(SMALLRYE_JWT_NEW_TOKEN_ISSUER)) {
+                devServiceProps.put(MP_JWT_VERIFY_ISSUER, DEFAULT_ISSUER);
+                devServiceProps.put(SMALLRYE_JWT_NEW_TOKEN_ISSUER, DEFAULT_ISSUER);
+            }
+
+            liveReloadBuildItem.setContextObject(KeyPairContext.class, new KeyPairContext(
+                    devServiceProps));
+
+            devServices.produce(smallryeJwtDevServiceWith(devServiceProps));
+        }
+
+        if (ctx != null && liveReloadBuildItem.isLiveReload()) {
+            devServices.produce(smallryeJwtDevServiceWith(ctx.properties()));
+        }
+    }
+
+    private Map<String, String> addDefaultSmallryePropertiesIfMissing(Set<String> userConfigs) {
+        HashMap<String, String> devServiceConfigs = new HashMap<>();
+        if (!userConfigs.contains(SMALLRYE_JWT_SIGN_KEY)) {
+            devServiceConfigs.put(SMALLRYE_JWT_SIGN_KEY, NONE);
+        }
+
+        if (!devServiceConfigs.containsKey(MP_JWT_VERIFY_PUBLIC_KEY)) {
+            devServiceConfigs.put(MP_JWT_VERIFY_PUBLIC_KEY, NONE);
+        }
+
+        return devServiceConfigs;
+    }
+
+    private boolean isConfigPresent(String property) {
+        return ConfigProvider.getConfig().getOptionalValue(property, String.class)
+                .isPresent();
+    }
+
+    private DevServicesResultBuildItem smallryeJwtDevServiceWith(Map<String, String> properties) {
+        return new DevServicesResultBuildItem(
+                Feature.SMALLRYE_JWT.name(), null, properties);
+    }
+
+    private static Map<String, String> generateDevServiceProperties(String publicKey, String privateKey) {
+        HashMap<String, String> properties = new HashMap<>();
+        properties.put(MP_JWT_VERIFY_PUBLIC_KEY, publicKey);
+        properties.put(SMALLRYE_JWT_SIGN_KEY, privateKey);
+        return properties;
+    }
+
+    private static String getStringKey(Key key) {
+        return Base64.getEncoder()
+                .encodeToString(key.getEncoded());
```

Comment:
I don't know that part of the codebase, but is this intended to be a PEM format?
RSA Key needs to use PKCS#1 format.

Reply:
Hi @cescoffier It is intended to be a PEM format. `mp.jwt.verify.publickey` uses it, [see here](https://pt.quarkus.io/guides/security-jwt#integration-testing-public-key).

Reply:
ok, this is **NOT** PEM format :-) This is just a base64 encoded key. 
If you want PEM format you need the header and footer explaining what's the content (RSA KEY, PRIVATE KEY ...)